src/buf_writer.c

```diff
@@ -112,10 +118,19 @@ buf_writer_cb(void *buf_writer_arg, const char *s) {
 	buf_writer_assert(buf_writer);
 	assert(buf_writer->buf != NULL);
 	assert(buf_writer->buf_end <= buf_writer->buf_size);
+	if (buf_writer->overflowed) {
+		return;
+	}
 	size_t i, slen, n, s_remain, buf_remain;
 	for (i = 0, slen = strlen(s); i < slen; i += n) {
 		if (buf_writer->buf_end == buf_writer->buf_size) {
-			buf_writer_flush(buf_writer);
+			if (buf_writer->allow_auto_flush) {
+				buf_writer_flush(buf_writer);
```

Comment:
Question: when reached here, we are sure buf_size is not 0, is that right?

Reply:
Yes. That's guarded by `assert(buf_writer->buf != NULL);` (`buf_size == 0` if and only if `buf == NULL`.)